### PR TITLE
config: fix Travis items to do more jdk11 and jdk12 builds

### PR DESCRIPTION
**Pull request by romani**
this is requirements to understand how to deal with #7033 .

it is refactoring and cleanup.
Main points:
1) we do not need jdk9 and jdk10, as jdk11 is available now and it will be our next JDK, we can use only latest jdk in non LTS jdk in travis to keep amount of builds limited.
2) `package -Passembly` do skip of tests as I use this only during release to generate "xxxxx-all.jar" 
3) `-Dlinkcheck.skip=true` it is execution of link check plugin that is running during `site` phase.
4) `\!` is bash is required only when it is used in command line directly, as we keep suppressions in special string constants it is not required. Escaping of it make regex be `\!` - so it simply does not work. I verified this by `-X` execution, and execution of `mvn -e package -Dtest=....` now actually executes tests and skip only defined.
5) `*,` in front of excludes is correct way to configure it - http://maven.apache.org/surefire/maven-surefire-plugin/examples/single-test.html , https://blog.jdriven.com/2017/10/run-one-or-exclude-one-test-with-maven/ , as it is property for what to run, NOT on what to exclude. So we run ALL except for items that are negated by `!`.

### CODE REVIEW

**Comment by romani** [comment]
@pbludov , PR is ready for review.

**Comment by pbludov** [comment]
Should we add JDK13 to the build matrix? Appveyor and Travis already support it. Some deprecated packages may have been dropped from JDK13.

**Comment by romani** [comment]
Ok, but as separate PR.
I would rather finish jdk12 update, before going further.

**Comment by romani** [comment]
@rnveach , ping

File: .ci/travis/travis.sh [hunk 1]
```diff
@@ -109,15 +105,20 @@ javac8)
...
-  mvn -e site -Dlinkcheck.skip=true
+jdk12-verify-limited)
+  # powermock doesn't support modifying final fields in JDK12, so need jacoco skip
+  exclude1="!FileContentsTest#testGetJavadocBefore,!FileTextTest#testFindLine*,"
+  exclude2="!MainFrameModelPowerTest#testOpenFileWithUnknownParseMode,"
+  exclude3="!TokenUtilTest#testTokenValueIncorrect2,"
+  exclude4="!ImportControlLoaderPowerTest#testInputStreamThatFailsOnClose"
+  # we skip pmd and spotbugs as they executed in special Travis build
+  mvn -e verify -Dtest=*,$exclude1$exclude2$exclude3$exclude4 -Djacoco.skip=true \
+    -Dpmd.skip=true -Dspotbugs.skip=true
```
Comment (rnveach) [inline .ci/travis/travis.sh]: Do we need to add a reason for jacoco skip?

**Pull request merged**